command_line/dlsnxs2cbf.py

```diff
@@ -0,0 +1,220 @@
+from __future__ import division, print_function
+from dxtbx.format import setup_hdf5_plugin_path
+
+setup_hdf5_plugin_path()
+import h5py
+from scitbx.array_family import flex
+
+sample = None
+datasets = []
+
+_mask = None
+
+
+def get_mask(nfast, nslow):
+    global _mask
+
+    if _mask:
+        return _mask
+
+    module_size_fast, module_size_slow = (1028, 512)
+    gap_size_fast, gap_size_slow = (12, 38)
+    n_fast, remainder = divmod(nfast, module_size_fast)
+    assert (n_fast - 1) * gap_size_fast == remainder
+
+    n_slow, remainder = divmod(nslow, module_size_slow)
+    assert (n_slow - 1) * gap_size_slow == remainder
+
+    mask = flex.bool(flex.grid(nslow, nfast), True)
+    blit = flex.bool(flex.grid(module_size_slow, module_size_fast), False)
+
+    for j in range(n_slow):
+        for i in range(n_fast):
+            o_i = i * (module_size_fast + gap_size_fast)
+            o_j = j * (module_size_slow + gap_size_slow)
+            mask.matrix_paste_block_in_place(blit, i_row=o_j, i_column=o_i)
+
+    _mask = mask
+    return _mask
+
+
+def depends_on(f):
+
+    global sample
+    global datasets
+
+    def finder(thing, path):
+        global datasets
+        datasets.append(path)
+        if hasattr(thing, "attrs"):
+            if thing.attrs.get("NX_class", None) == "NXsample":
+                global sample
+                sample = path
+
+        if hasattr(thing, "keys"):
+            for k in thing:
+                try:
+                    finder(thing[k], path="%s/%s" % (path, k))
+                except (IOError, TypeError, ValueError, KeyError):
+                    pass
+
+    finder(f, path="")
+
+
+def get_distance_in_mm(f):
+    try:
+        D = f["/entry/instrument/detector_distance"]
+    except KeyError:
+        D = f["/entry/instrument/detector/detector_distance"]
+    d = D[()]
+    if D.attrs["units"] == "m":
+        d *= 1000
+    elif D.attrs["units"] == "mm":
+        pass
+    else:
```

Comment:
```suggestion
    elif D.attrs["units"] != "mm":
```
Changing this `else` statement as above would eliminate the need for the `elif...pass` statement above it.

Reply:
👍